encoding/proto.go

```diff
@@ -0,0 +1,28 @@
+package encoding
+
+import (
+	"errors"
+
+	"google.golang.org/protobuf/proto"
+)
+
+// PBcodec encodes/decodes Go values to/from protocol buffers.
+type PBcodec struct{}
+
+// Marshal encodes a Go value to protocol buffers.
+func (c PBcodec) Marshal(v interface{}) ([]byte, error) {
+	msg, ok := v.(proto.Message)
```

Comment:
The other codecs, including two that aren't merged yet ([TOML](https://github.com/philippgille/gokv/pull/103), [MessagePack](https://github.com/philippgille/gokv/pull/98)) allow the marshalling of any Go type (string, slices, structs) and there are `gokv` store tests that test this: [Convenience function in test package](https://github.com/philippgille/gokv/blob/93d491e1a0c9ba21a1a88e7e5888da159671a122/test/test.go#L91), [example usage in a store test](https://github.com/philippgille/gokv/blob/93d491e1a0c9ba21a1a88e7e5888da159671a122/consul/consul_test.go#L41-L57)

This codec here would only allow passing objects of proto messages, right?

I'm not sure if it's maybe too much of a restriction for general use. I agree that protobufs are very popular and widely used (incl by myself), so it could be a nice convenience. But then we should probably document this constraint very clearly, and maybe another reason to organize codecs in subdirectories which might decrease the expectation that all codecs work in a similar way.

WDYT?

Reply:
No I don't think there's a way around it.

Can you change the Godoc from "Marshal encodes a Go value to protocol buffers." to something that more clearly indicates the restriction? I.e. the passed value can't be any Go value, but must be an object of a proto message struct.

Reply:
sure, done